File: OOP/encapsulation.py

# This is public encapsulation
class Student:

        roll = 39829832 
        
        def __init__(self,roll):
                self.roll = roll

        def display(self):
                return(self.roll)
        

student = Student(2006015)
print(student.display())

student.roll = 2006016 #Here, trying to change the roll, it's possible because of public encapsulation
print(student.display())



# Private encapsulationo

class Teacher:
        _teacherId = 0
        def __init__(self, teacherID):
                self._teacherId = teacherID

        def display(self):
                return (self._teacherId)
        
teacher = Teacher(50)
print(teacher.display())

teacher._teacherId = 60
print(teacher.display())


class Student:
   def __init__(self, roll):
       self.roll = roll

File: OOP/encapsulation.cpp

// Public encapsulation


#include<bits/stdc++.h>
using namespace std;

class Student{
        public:
        int roll;
        
        Student(int roll){
                this->roll = roll;
        }

        void display(){
                cout << roll << endl;
        }
};



int main(){
        Student student(2006015);
        student.display();
        student.roll = 2006030;
        student.display();

        return 0;
}


// Private incapsulation

#include<bits/stdc++.h>
using namespace std;

class Student{
        private:
           int roll;

        public:
        
           Student(int roll){
                this->roll = roll;
        }

        void setter(int r){
                roll = r;
        }

        int getter(){
                return roll;
        }
};



int main(){
        Student student(2006015);
        student.setter(2006013);
        cout << student.getter() << endl;

        return 0;
}
